js/gated-content/src/components/VideoListing.vue

```diff
@@ -45,6 +52,34 @@ export default {
         throw error;
       });
   },
+  methods: {
+    // TODO: maybe we need to move this method to mixin?
+    // (note: this can be singe or multiple values, compare LiveStreamListing and LiveStreamPage)
+    combine(data) {
+      if (!data.included) return;
+      this.listing.forEach((video, key) => {
```

Comment:
I think we have to create a helper fucntion/class and export it to the separate file.
This will help us to reuse this code and improve readability.
Example: https://github.com/ymcatwincities/openy_cycle_health/blob/master/docroot/modules/custom/twelve_app/js/helper/twelve/user.js